File: SQLAlchemy/readsnowflate/df.py
from sqlalchemy import create_engine, text
import pandas as pd


# Contoh penggunaan:
username = 'SILVIO'
password = 'S1LV1Oooo'
account_identifier = 'rf60339.ap-southeast-1'
database = 'SNOWFLAKE_SAMPLE_DATA'
schema = 'TPCDS_SF100TCL'
warehouse = 'COMPUTE_WH'
role = 'ACCOUNTADMIN'
# Membuat URL koneksi
engine = create_engine(
    f"snowflake://{username}:{password}@{account_identifier}/{database}/{schema}?warehouse={warehouse}&role={role}"
)

# # Membuat URL koneksi
# engine = create_engine(
#     'snowflake://SILVIO:S1LV1Oooo@rf60339.ap-southeast-1/SNOWFLAKE_SAMPLE_DATA/TPCDS_SF100TCL?warehouse=COMPUTE_WH&role=ACCOUNTADMIN'
# )

# Query SQL
query = text("SELECT * FROM CATALOG_PAGE")

# Eksekusi query menggunakan engine dan simpan hasil ke DataFrame
result_proxy = engine.execute(query)
df = pd.DataFrame(result_proxy.fetchall(), columns=result_proxy.keys())

# Menampilkan DataFrame
print(df)

# Tutup koneksi setelah selesai
engine.dispose()  # Close all connections in the connection pool

File: akses sql dengan R/readsnowflake/df.r
library(DBI)
library(odbc)

# Membuat koneksi
con <- dbConnect(odbc::odbc(),
                 Driver = "SnowflakeDSIIDriver",
                 Server = "rf60339.ap-southeast-1.snowflakecomputing.com",
                 UID = "SILVIO",
                 PWD = "S1LV1Oooo",
                 Database = "SNOWFLAKE_SAMPLE_DATA",
                 Schema = "TPCDS_SF100TCL",
                 Warehouse = "COMPUTE_WH",
                 Role = "ACCOUNTADMIN")

# Menjalankan query
result <- dbGetQuery(con, "SELECT * FROM CATALOG_PAGE")

# Mengubah hasil query menjadi data frame
result_df <- as.data.frame(result)
print(result_df)
# Menampilkan hasil
print(head(result_df))

# Menutup koneksi
dbDisconnect(con)
